snow/engine/avalanche/bootstrap/bootstrapper.go

```diff
@@ -336,38 +329,38 @@ func (b *bootstrapper) Start(ctx context.Context, startReqID uint32) error {
 		return fmt.Errorf("failed to get linearization status: %w", err)
 	}
 	if linearized {
-		edge := b.Manager.Edge(ctx)
-		return b.ForceAccepted(ctx, edge)
+		return b.ForceAccepted(ctx, nil)
 	}
 
-	// If requested, assume the currently accepted state is what was linearized.
-	//
-	// Note: This is used to linearize networks that were created after the
-	// linearization occurred.
-	if b.Config.LinearizeOnStartup {
-		edge := b.Manager.Edge(ctx)
-		stopVertex, err := b.Manager.BuildStopVtx(ctx, edge)
-		if err != nil {
-			return fmt.Errorf("failed to create stop vertex: %w", err)
-		}
-		if err := stopVertex.Accept(ctx); err != nil {
-			return fmt.Errorf("failed to accept stop vertex: %w", err)
-		}
-
-		stopVertexID := stopVertex.ID()
-		b.Ctx.Log.Info("accepted stop vertex",
-			zap.Stringer("vtxID", stopVertexID),
+	// If a stop vertex is well known, accept that.
+	if b.Config.StopVertexID != ids.Empty {
+		b.Ctx.Log.Info("using well known stop vertex",
+			zap.Stringer("vtxID", b.Config.StopVertexID),
 		)
 
-		return b.ForceAccepted(ctx, []ids.ID{stopVertexID})
+		return b.ForceAccepted(ctx, []ids.ID{b.Config.StopVertexID})
 	}
 
-	if !b.StartupTracker.ShouldStart() {
-		return nil
+	// If a stop vertex isn't well known, treat the current state as the final
+	// DAG state.
+	//
+	// Note: This is used to linearize networks that were created after the
+	// linearization occurred.
+	edge := b.Manager.Edge(ctx)
+	stopVertex, err := b.Manager.BuildStopVtx(ctx, edge)
+	if err != nil {
+		return fmt.Errorf("failed to create stop vertex: %w", err)
+	}
+	if err := stopVertex.Accept(ctx); err != nil {
+		return fmt.Errorf("failed to accept stop vertex: %w", err)
 	}
 
-	b.started = true
-	return b.Startup(ctx)
+	stopVertexID := stopVertex.ID()
+	b.Ctx.Log.Info("generated stop vertex",
+		zap.Stringer("vtxID", stopVertexID),
+	)
+
+	return b.ForceAccepted(ctx, nil)
```

Comment:
q: why is this not `b.ForceAccepted(ctx, []ids.ID{stopVertexID})`?
IIUC with b.ForceAccepted(ctx, nil) will start downloading missing vertexes, while stopVertexID is guaranteed to be already available locally

Reply:
I see that in ForceAccepted we append the vertexs found there with those missing already stored locally. So I understand that ForceAccepted may download something even if we pass ni here.
Also my question is: do we want to guarantee that a node download all accepted vertexes?

Reply:
We do (and must) guarantee that a node downloads all accepted vertices